### ISSUE TITLE
Regression in `DeclareHandlingByNiceBasis`/`InstallHandlingByNiceBasis`

### ISSUE THREAD

**Issue by zickgraf** [gap-system/gap#5322]
The documentation of `DeclareHandlingByNiceBasis`/`InstallHandlingByNiceBasis` claims:

> name must be a string, a filter f with this name is created, and a logical implication from f to IsHandledByNiceBasis (61.11-6) is installed.

This is not true anymore since commit 2f4a518d36912392d7102c940873e77b2c076c23 (see https://github.com/gap-system/gap/commit/2f4a518d36912392d7102c940873e77b2c076c23#diff-14aba9b3712179af58d64fc6572ca6107a7c746245c8c795a98b79ec746676d4R690-L694).

This leads to an error when loading [QPA2](https://github.com/sunnyquiver/QPA2):
```
Error, required filters [
    "IsSubspaceOfQPAVectorSpace" 
]
for 1st argument do not match a declaration of NiceFreeLeftModule called from
<<compiled GAP code>> from GAPROOT/lib/oper1.g:336 in function InstallMethod called from
<function "InstallMethod">( <arguments> )
 called from read-eval loop at QPA2/lib/vecspace.gi:766
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
```

I'm not sure if the documentation or the regressing commit should be updated.

**Comment by ThomasBreuer** [comment]
@zickgraf Yes, the new filter `f` is now just a flag such that `f and IsFreeLeftModule and IsAttributeStoringRep` implies `IsHandledByNiceBasis`. I think this is an improvement over the situation before #3006 got merged, and I am going to update the documentation accordingly.

The mentioned problem in the QPA2 package arises from the fact that the `NiceFreeLeftModule` method wants to be applicable as soon as the argument is in `IsSubspaceOfQPAVectorSpace`, which does no longer imply `IsFreeLeftModule`.
(Most of the vector spaces types in the GAP library use the generic `NiceFreeLeftModule` method, that's why I had overlooked this problem.)
If it is acceptable to adjust the QPA2 code then I would suggest to change the `InstallMethod` call such that `IsFreeLeftModule and IsSubspaceOfQPAVectorSpace` is required.
